### PR TITLE
Deprecation for resources

### PR DESCRIPTION
**Pull request by tfili**
A bunch of places now take a `Resource` as a url. Some of these classes also took other parameters that are now part of the resource (eg. proxy, headers, query). These parameters have been deprecated since they can be passed in as part of the resource instance.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@tfili, thanks for the pull request! Maintainers, we have a signed CLA from @tfili, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Renderer/loadCubeMap.js [hunk 1]
```diff
@@ -66,6 +68,10 @@ define([
         }
         //>>includeEnd('debug');
 
+        if (defined(allowCrossOrigin)) {
+            deprecationWarning('loadCubeMap.allowCrossOrigin', 'The allowCrossOrigin parameter has been deprecated. It no longer needs to be specified.');
```
Comment (mramato) [inline Source/Renderer/loadCubeMap.js]: This function is private, so we shouldn't need to deprecate anything.

File: Source/Scene/ModelInstanceCollection.js [hunk 2]
```diff
@@ -106,6 +107,10 @@ define([
         }
         //>>includeEnd('debug');
 
+        if (defined(options.headers)) {
```
Comment (mramato) [inline Source/Scene/ModelInstanceCollection.js]: This is private as well, so we shouldn't need this deprecation.

**Comment by mramato** [comment]
Just those comments, nothing else jumps out at me (and this will get another look in the main PR) so once you address them I'll merge (or you can just merge yourself since this goes into a branch).

**Pull request merged**